<<<<<<< HEAD
From: Felix Fietkau <nbd@nbd.name>
Date: Wed, 20 Oct 2021 21:13:10 +0200
Subject: [PATCH] hostpad: fix a race condition on adding AP mode wds sta
 interfaces

Both hostapd and netifd attempt to add a VLAN device to a bridge.
Depending on which one wins the race, bridge vlan settings might be incomplete,
or hostapd might run into an error and refuse to service the client.
Fix this by preventing hostapd from adding interfaces to the bridge and
instead rely entirely on netifd handling this properly

--- a/hostapd/config_file.c
+++ b/hostapd/config_file.c
@@ -2472,6 +2472,8 @@ static int hostapd_config_fill(struct ho
=======
--- a/hostapd/config_file.c
+++ b/hostapd/config_file.c
@@ -2405,6 +2405,8 @@ static int hostapd_config_fill(struct ho
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
 			   sizeof(conf->bss[0]->iface));
 	} else if (os_strcmp(buf, "bridge") == 0) {
 		os_strlcpy(bss->bridge, pos, sizeof(bss->bridge));
+		if (!bss->wds_bridge[0])
+			os_strlcpy(bss->wds_bridge, pos, sizeof(bss->wds_bridge));
<<<<<<< HEAD
 	} else if (os_strcmp(buf, "bridge_hairpin") == 0) {
 		bss->bridge_hairpin = atoi(pos);
 	} else if (os_strcmp(buf, "vlan_bridge") == 0) {
--- a/src/ap/ap_drv_ops.c
+++ b/src/ap/ap_drv_ops.c
@@ -394,8 +394,6 @@ int hostapd_set_wds_sta(struct hostapd_d
=======
 	} else if (os_strcmp(buf, "vlan_bridge") == 0) {
 		os_strlcpy(bss->vlan_bridge, pos, sizeof(bss->vlan_bridge));
 	} else if (os_strcmp(buf, "wds_bridge") == 0) {
--- a/src/ap/ap_drv_ops.c
+++ b/src/ap/ap_drv_ops.c
@@ -340,8 +340,6 @@ int hostapd_set_wds_sta(struct hostapd_d
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
 		return -1;
 	if (hapd->conf->wds_bridge[0])
 		bridge = hapd->conf->wds_bridge;
-	else if (hapd->conf->bridge[0])
-		bridge = hapd->conf->bridge;
 	return hapd->driver->set_wds_sta(hapd->drv_priv, addr, aid, val,
 					 bridge, ifname_wds);
 }
